src/cookies.js

```diff
@@ -100,14 +100,15 @@ export function setCookie(win, name, value, expirationTime, opt_options) {
   if (opt_options && opt_options.domain) {
     domain = opt_options.domain;
   } else if (opt_options && opt_options.highestAvailableDomain) {
-    domain = getHighestAvailableDomain(win);
+    domain = /** @type {string} */ (getHighestAvailableDomain(win));
   }
   trySetCookie(win, name, value, expirationTime, domain);
 }
 
 /**
  * Attemp to find the HighestAvailableDomain on
  * @param {!Window} win
+ * @return {?string}
```

Comment:
Can this return `{string}` so callers don't have to cast?

Reply:
Not possible because `null` is an intended return value.

https://github.com/ampproject/amphtml/blob/569d20852833dd1a47e4296ea6354989aa4017ce/src/cookies.js#L154-L162

Reply:
Can you assert not-null instead?

Reply:
This requires knowledge of the inner workings of this code, to determine whether it should be a `dev()` or `user()` level assert, which is out of scope for this PR.